### DISCUSSION TITLE
Upgrading from 20.04lts to 22.04lts

### DISCUSSION THREAD

**Discussion by MikeandRosanna** [linux-surface/linux-surface#917] [Q&A]
Hi, I have Ubuntu 20.04 lts with Linux-Surface 5.19.4 running happily on my Surface Laptop 3. My question is whether upgrading to Ubuntu 22.04 lts using the Ubuntu upgrade button would mean  losing the Linux-Surface kernel and having to start all over again?
Any answers appreciated.

**Comment by NP-chaonay** [comment]
I dont know about this but If I guess it correct

1) oldkernel is existed and it should able to do boot
2) linux-surface repo is temporarily disabled on upgrade, you enable it later and then do upgrade and it should have new linux-surface kernel
3) distro upgrade is just big package upgrade and distro info changed with user-added repo disabled, so i dont think it require new setup, unless you see it needed.

But what I'm afriad is that 
1) the package manager decied to uninstall pkgs installed from linux-surface wiki due to dependency problem that the reuqired pkg no longer in new distro version
2) the big update breaks the config and setting of linux-surface stuff (but you dont have to worry about this much, since the distro upgrade is similar to apt upgrade, so even the normal upgrade, it can break it too.)

**Comment by MikeandRosanna** [comment]
Hi Nuttapong Punpipat,
Many thanks for your speedy reply.
Regards,
Michael Murphy.
